src/structures/GuildChannel.js

```diff
@@ -227,15 +228,19 @@ class GuildChannel extends Channel {
    *   .then(channel => console.log(channel.permissionOverwrites.get(message.author.id)))
    *   .catch(console.error);
    */
-  async updateOverwrite(userOrRole, options, reason) {
-    userOrRole = this.guild.roles.resolve(userOrRole) || this.client.users.resolve(userOrRole);
-    if (!userOrRole) return Promise.reject(new TypeError('INVALID_TYPE', 'parameter', 'User nor a Role'));
+  async updateOverwrite(userOrRole, options, reason, type) {
```

Comment:
might it also actually be worth combining both `reason` and `type` into a single parameter w/ an object? because both are optional

Reply:
Sure, just need some time to come up with a good name and a night's sleep :) 👍

Reply:
Added the suggested changes in 0bacaa8. Just want to make sure they are okay before I add the typings for them.